avm/res/network/public-ip-prefix/main.bicep

```diff
@@ -26,6 +26,13 @@ param tags object?
 @description('Optional. The custom IP address prefix that this prefix is associated with. A custom IP address prefix is a contiguous range of IP addresses owned by an external customer and provisioned into a subscription. When a custom IP prefix is in Provisioned, Commissioning, or Commissioned state, a linked public IP prefix can be created. Either as a subset of the custom IP prefix range or the entire range.')
 param customIPPrefix object = {}
 
+@description('Optional. A list of availability zones denoting the IP allocated for the resource needs to come from.')
```

Comment:
Please update as per [BCPFR5 - Category: Inputs - Availability Zones Implementation](https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpfr5---category-inputs---availability-zones-implementation).

To ease usuability we recommend to use 'primitive types' where possible. There is also a [PR](https://github.com/Azure/Azure-Verified-Modules/pull/803) open to re-enforce this.

Reply:
Has this been successfully implemented yet?

In this case following implementation crashes the Set-AVMmodule:

```bicep
@description('Optional. A list of availability zones denoting the IP allocated for the resource needs to come from.')
@allowed([
  1
  2
  3
])
param zones int[] = [
  1
  2
  3
]

resource publicIpPrefix 'Microsoft.Network/publicIPPrefixes@2023-09-01' = {
  name: name
  location: location
  tags: tags
  sku: {
    name: 'Standard'
  }
  zones: [for zone in zones: string(zone)]
  properties: {
    customIPPrefix: !empty(customIPPrefix) ? customIPPrefix : null
    publicIPAddressVersion: 'IPv4'
    prefixLength: prefixLength
  }
}
```

But it works when I use a variable, like this:

```bicep
@description('Optional. A list of availability zones denoting the IP allocated for the resource needs to come from.')
@allowed([
  1
  2
  3
])
param zones int[] = [
  1
  2
  3
]

var zoneStrings = [for zone in zones: string(zone)]

resource publicIpPrefix 'Microsoft.Network/publicIPPrefixes@2023-09-01' = {
  name: name
  location: location
  tags: tags
  sku: {
    name: 'Standard'
  }
  zones: zoneStrings
  properties: {
    customIPPrefix: !empty(customIPPrefix) ? customIPPrefix : null
    publicIPAddressVersion: 'IPv4'
    prefixLength: prefixLength
  }
}
```

Reply:
Just ran some test. From a template perspective it looks fine and doesn't show an error, yet when I try to deploy Bicep get's stuck upon depoyment. It just doesn't do anything. Looks to me like a Bicep bug for which an issue should probably be created with the same content that you posted above (working vs. non-working) example.

That being said, if the variable unblocks you, let's add it. But we should then also update the spec so others are not running into the same issue. Could you do that once you finished this PR and we're sure the suggestion works?

Reply:
AVM Spec update: https://github.com/Azure/Azure-Verified-Modules/pull/813

Reply:
@krbar I'm currently testing a [suggestion](https://github.com/Azure/bicep/issues/9668#issuecomment-2035405301) made by @jeskew via Pull Request #1521

If it works as hoped, we should apply the implementation here too and update the spec again

Reply:
Alright @krbar, this `map(zones, zone => string(zone))` does the trick as it does not result in a copy loop, but uses the new lambda functions.